### PR TITLE
Add add_test as webnn delegate unit test example

### PR DESCRIPTION
**Pull request by zesongw**
Test with all add cases: bazel test  --test_output=all //tensorflow/lite/delegates/webnn:add_test

### CODE REVIEW

**Comment by Honry** [comment]
@zesongw, I rebased the webnn delegate to latest webnn-native at PR https://github.com/huningxin/tensorflow/pull/8, could you retest these unit tests based on latest commit?

File: tensorflow/lite/delegates/webnn/add_test.cc [hunk 1]
```diff
@@ -0,0 +1,935 @@
...
+      .Input2Shape({1, 1, 1, channels})
+      .Test(BuiltinOperator_ADD, webnn_delegate.get());
+}
+
+TEST(Add, 4DBy4DBroadcastWidth) {
+  TfLiteWebNNDelegateOptions delegate_options =
+      TfLiteWebNNDelegateOptionsDefault();
+  std::unique_ptr<TfLiteDelegate, decltype(&TfLiteWebNNDelegateDelete)>
+      webnn_delegate(TfLiteWebNNDelegateCreate(&delegate_options),
+                       TfLiteWebNNDelegateDelete);
```
Comment (Honry) [inline tensorflow/lite/delegates/webnn/add_test.cc]: Nit: indentation. Please check and fix the rest indentation issues.

```suggestion
                     TfLiteWebNNDelegateDelete);
```

File: tensorflow/lite/delegates/webnn/add_test.cc [hunk 2]
```diff
@@ -0,0 +1,935 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
```
Comment (Honry) [inline tensorflow/lite/delegates/webnn/add_test.cc]: Let's use Copyright 2022 for all new files.
```suggestion
/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
```

File: tensorflow/lite/delegates/webnn/add_test.cc [hunk 3]
```diff
@@ -0,0 +1,935 @@
...
+      .Input2Shape({batch, height, width, channels})
+      .Test(BuiltinOperator_ADD, webnn_delegate.get());
+}
+
+TEST(Add, 4DBy4DBroadcastChannels) {
+  TfLiteWebNNDelegateOptions delegate_options =
+      TfLiteWebNNDelegateOptionsDefault();
+  std::unique_ptr<TfLiteDelegate, decltype(&TfLiteWebNNDelegateDelete)>
+      webnn_delegate(TfLiteWebNNDelegateCreate(&delegate_options),
+                       TfLiteWebNNDelegateDelete);
```
Comment (Honry) [inline tensorflow/lite/delegates/webnn/add_test.cc]: Nit: indentation.

```suggestion
                     TfLiteWebNNDelegateDelete);
```

File: tensorflow/lite/delegates/webnn/add_test.cc [hunk 4]
```diff
@@ -0,0 +1,935 @@
...
+  const auto channels = shape_rng();
+
+  BinaryElementwiseTester()
+      .Input1Shape({batch, height, width, channels})
+      .Input2Shape({batch, height, width, channels})
+      .SignBitActivation()
+      .Test(BuiltinOperator_ADD, webnn_delegate.get());
+}
+
+// TEST(Add, MultiThreading) {
```
Comment (Honry) [inline tensorflow/lite/delegates/webnn/add_test.cc]: Please remove these notations, webnn doesn't support multiThreading.

**Review by Honry — COMMENTED** [review]
Thanks @zesongw, LGTM % some nits.

File: tensorflow/lite/delegates/webnn/add_test.cc [hunk 5]
```diff
@@ -44,15 +45,15 @@ TEST(Add, 4DBy4D) {
   BinaryElementwiseTester()
       .Input1Shape({batch, height, width, channels})
       .Input2Shape({batch, height, width, channels})
-      .Test(BuiltinOperator_ADD, webnn_delegate.get());
+      .Test(BuiltinOperator_MUL, webnn_delegate.get());
```
Comment (Honry) [inline tensorflow/lite/delegates/webnn/add_test.cc]: Nit: 
```suggestion
      .Test(BuiltinOperator_ADD, webnn_delegate.get());
```

**Comment by Honry** [comment]
@huningxin, PTAL as well, thanks!

**Review by Honry — APPROVED** [review]
LGTM, thanks!

**Review by huningxin — APPROVED** [review]
LGTM, thanks @zesongw .

**Pull request merged**